golang/cosmos/app/app.go

```diff
@@ -825,13 +825,25 @@ func NewAgoricApp(
 	return app
 }
 
+func ensureBlockMaxBytes(app *GaiaApp, ctx sdk.Context) {
+	cp := app.BaseApp.GetConsensusParams(ctx)
+	if cp.Block.MaxBytes > gaiaappparams.MaxBlockMaxBytes {
+		bp := *cp.Block
+		bp.MaxBytes = gaiaappparams.MaxBlockMaxBytes
+		cp.Block = &bp
```

Comment:
`cp` and `cp.Block` are pointers, so you ought to be able to just do

```suggestion
		cp.Block.MaxBytes = gaiaappparams.MaxBlockMaxBytes
```

Reply:
The `Block` is coming directly from the store. I hadn't tracked down if was acceptable to modify in place or not, but apparently the `Subspace` implementation backing the `ParamStore` goes through a marshalling phase, so it should be fine.